### DISCUSSION TITLE
MGDM 129.1 et Biol_oekol_Gesamtbeurteilung

### DISCUSSION THREAD

**Discussion by TriformMoritz** [QGEP/QGEP#700] [Ideas]
Hello, the confederation is demanding (via the cantons) the MGDM for PGEE, identificateur 129.1 for example [https://www.vs.ch/web/sen/evacuation-des-eaux], click on "Identificateur 129.1" to download the ili model and the description.
They are a lot of attributes from the VSA-DSS models who could/should  be derived directly from our QGEP model.

Questions:

- Are we planning to integrate views an export to this MGDM model?
- Where can I find the "Biol_oekol_Gesamtbeurteilung", which is part of the VSA-DSS and necessary for the export?

Proposal:

- Either create the necessary views or create an export (xtf).

I started a SQL myself for StrukturZustand (virtual layer in QGIS), sorry for my bad programming:

 SELECT 1 as all_id, 
	round(sum(ca.surface_area),0) as FIEntw,
	round(ARA.FIRedARA,0)as FIRedARA,
	round(RedG.FIRedGewaesser,0) as FIRedGewaesser,
	round(trenn.FlTrennsystem,0)as FlTrennsystem,
	round(misch.FlMischsystem,0)as FlMischsystem,
	round(vers.FIVers,0)as FIVers,
	round(Koeff.LaengeKanalOeff,0)as LaengeKanalOeff,
	round(Kpriv.LaengeKanalPrive,0)as LaengeKanalPrive,
	round(unto.LaengeUntersuchtOeff,0)as LaengeUntersuchtOeff,
	round(zusto.LaengeZustandOeff,0)as LaengeZustandOeff,
	round(untp.LaengeUntersuchtPriv,0)as LaengeUntersuchtPriv,
	zustp.LaengeZustandPriv,
	'manuel' as VolNetzMWbehandl,
	AnzEin.AnzEinleitst,
	'manuel' as AnzEinleitstUntersucht,
	'manuel' as AnzEinleitstBedarf,
	'manuel' as FWMenge 
FROM catchment_area ca
JOIN(
	SELECT  1 as ID, sum(surface_area*seal_factor_rw_current/100) as FIRedARA
	FROM catchment_area
	WHERE drainage_system_current = 5186) ARA
ON ARA.id = all_id
JOIN(
	SELECT 1 as ID, sum(surface_area*seal_factor_rw_current/100) as FIRedGewaesser
	FROM catchment_area
	WHERE drainage_system_current = 5187) RedG
On RedG.id = all_id
JOIN(
	SELECT 1 as ca_id, sum(ca2.surface_area) as FlMischsystem
	FROM 
	catchment_area ca2
	WHERE drainage_system_current = 5186) misch
ON misch.ca_id = all_id
JOIN(
	SELECT 1 as id, sum(ca2.surface_area) as FlTrennsystem
	FROM 
	catchment_area ca2
	WHERE drainage_system_current = 5187) trenn
ON trenn.id = all_id
JOIN(
	SELECT 1 as id, sum(surface_area) as FIVers
	FROM catchment_area 
	WHERE infiltration_current = 5452) vers
ON vers.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeKanalOeff
	FROM vw_qgep_troncon
	WHERE "ws_financing" = 5510) Koeff
ON Koeff.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeKanalPrive
	FROM vw_qgep_troncon
	WHERE "ws_financing" = 5511) Kpriv
ON Kpriv.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeUntersuchtOeff
	FROM vw_qgep_troncon
	WHERE  "ws_structure_condition" <> 3037 and "ws_financing" = 5510 ) unto
ON unto.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeZustandOeff 
	FROM vw_qgep_troncon
	WHERE  "ws_structure_condition" in(3363,3359) and "ws_financing" = 5510 ) zusto
ON zusto.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeUntersuchtPriv
	FROM vw_qgep_troncon
	WHERE  "ws_structure_condition" <> 3037 and "ws_financing" = 5511 ) untp
ON untp.id = all_id
JOIN(
	SELECT 1 as id, sum(length_effective) as LaengeZustandPriv
	FROM vw_qgep_troncon
	WHERE  "ws_structure_condition" in(3363,3359) and "ws_financing" = 5511) zustp
ON zustp.id = all_id
JOIN(
	SELECT 1 as id, count() as AnzEinleitst
	FROM Ouvrages
	WHERE  "ws_type" ='discharge_point' AND "dp_relevance" = 5580 ) AnzEin
ON AnzEin.id = all_id

**Comment by sjib** [comment]
"Biol_oekol_Gesamtbeurteilung" is part of the extension Stammkarte in the VSA-DSS 2015 model release. We have been thinking about implementing this for a long time, but have not done so yet. The question is, whether we do it on the release 2015 version or directly add the 2020 version for these three classes (Stammkarte, Biol_oekol_Gesamtbeurteilung and Gesamteinzugsgebiet). This depends, whether we want to be able to export it in the 2015 release or if we implement INTERLIS export for that when we have changed to 2020. What do you think? Please add your preferences.

**Reply by TriformMoritz** [reply]
In my case, it would be easier to have it in a stable 2015 release with extension and than Interlis export /import. Then switch to the 2020 release. Will up and downgrading of the releases (VSA2015 -> VSA2020 and VSA2020 -> VSA2015) be possible? For example in Interlis via the GEP-Datachecker?

**Comment by sjib** [comment]
I would create a new view for that in the database instead of a virtual layer:

`CREATE OR REPLACE VIEW qgep_od.vw_mdgm_StrukturZustand 
 AS`

and then reference to the tables by qgep_od.* 
Then it is also not language specific (as eg. vw_qgep_troncon -> qgep_od.vw_qgep_reach)

What is the layer ARA?

The JOIN statements do not look proper:

`JOIN(
SELECT 1 as ca_id, sum(ca2.surface_area) as FlMischsystem
FROM
catchment_area ca2
WHERE drainage_system_current = 5186) misch
ON misch.ca_id = all_id`

The `) misch` does not make sense at that place - what do you want to name with `misch`

Different types of JOIN are described e.g. [here](https://www.tutorialspoint.com/postgresql/postgresql_using_joins.htm)
